### ISSUE TITLE
Using "first" for indent can result in message about "NaN spaces" with nested arrays

### ISSUE THREAD

**Issue by Trott** [eslint/eslint#7727]
**Tell us about your environment**

* **ESLint Version:** current master 
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
config file is empty

```
```

**What did you do? Please include the actual source code causing the issue.**

Running command:

```sh
eslint --rule 'indent: [2, 2, {ArrayExpression: first}]' test.js
```

with contents of `test.js` as follows:

```js
var foo = [
  [
     1, 2, 3,
     4, 5, 6
  ]
];
```

**What did you expect to happen?**

Either no error or else expected indentation of some positive integer.

**What actually happened? Please include the actual, raw output from ESLint.**

  2:3  error  Expected indentation of NaN spaces but found 2  indent
  5:3  error  Expected indentation of NaN spaces but found 2  indent

**Comment by Trott** [comment]
@not-an-aardvark I imagine there may be a release planned for tomorrow. If so, I imagine you'll want to fix this issue (or revert the new feature) before releasing.
